tests/lib/min_heap/src/test_min_heap.c

```diff
@@ -0,0 +1,154 @@
+/**
+ * @file test_min_heap.c
+ * @brief Min-heap tests using ztest framework
+ *
+ * This file contains unit tests for verifying the functionality of
+ * the min-heap implementation in Zephyr. It uses `ztest` and demonstrates
+ * heap initialization, insertion, ordering, popping and destroying.
+ */
+
+/*
+ * Copyright (c) 2025 Aerlync Labs Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/sys/min_heap.h>
+
+/** @brief Maximum number of items in the test heap */
+#define MAX_ITEMS 10
+
+/** @brief Test item structure with heap node and identifier string*/
+struct test_item {
+	/** Min heap node containing the key */
+	struct min_heap_node node;
+	/** Name for the test item */
+	const char *name;
+	/** Key for comparison */
+	int key;
+};
+
+/** @brief Buffer to hold heap node pointers */
+static struct min_heap_node *heap_buf[MAX_ITEMS];
+/** @brief Min heap instance for testing */
+static struct min_heap heap;
+
+/** @brief Predefined set of test items with associated priorities */
+static struct test_item items[MAX_ITEMS] = {
+	{ .name = "Task5", .key = 5 },
+	{ .name = "Task1", .key = 1 },
+	{ .name = "Task3", .key = 3 },
+	{ .name = "Task7", .key = 7 },
+	{ .name = "Task2", .key = 2 },
+	{ .name = "Task8", .key = 8 },
+	{ .name = "Task4", .key = 4 },
+	{ .name = "Task0", .key = 0 },
+	{ .name = "Task6", .key = 6 },
+	{ .name = "Task9", .key = 9 }
+};
+
+/**
+ * @brief Comparator function for min-heap nodes
+ *
+ * @param a First heap node
+ * @param b Second heap node
+ *
+ * @return Negative if a < b, 0 if equal, positive if a > b
+ */
+static int test_cmp(const struct min_heap_node *a,
+		    const struct min_heap_node *b)
+{
+	const struct test_item *item_a = CONTAINER_OF(a,
+					 struct test_item, node);
+	const struct test_item *item_b = CONTAINER_OF(b,
+					 struct test_item, node);
+
+	return item_a->key - item_b->key;
+}
+
+MIN_HEAP_DEFINE(test_heap, MAX_ITEMS, test_cmp);
+
+/**
+ * @brief Test ordering of items inserted into min-heap
+ *
+ * Inserts predefined items into the heap and asserts that
+ * popping the heap returns them in ascending order of key.
+ */
+ZTEST(min_heap_suite, test_min_heap_ordering)
```

Comment:
Can tests be made for each API function?

E.g. test_min_heap_init, test_min_heap_insert, test_min_heap_remove, test_min_heap_push, test_min_heap_pop, test_min_heap_find, etc?

Tests are usually executed in alphabetical order as well, so it can be helpful to order them that way in the file.

It may help to have a `test_min_heap_init_common()` function that is called into for verifying that the macro-based initializers also initialize a heap correctly.

I would also suggest reinitializing the min heap before each test in a `before()` function.